programs/utilities/keepalive.sc

```diff
@@ -1,21 +1,59 @@
 // re-logs fake players upon server restart
 // must need to place it directly in the world scripts folder
 
-__config() -> {'scope' -> 'global'};
+__config() -> {
+	requires -> {
+		'minecraft' -> '>=1.8', // spectator was not a thing prior to 1.8
+	},
+	'scope' -> 'global'
+};
 
+__on_connect(e) -> (
+	// forcing the fake-players that were spawned in spectator back into creative to retain flight per their data
+	if(e:'gm' == 'creative' && e:'fly' == 1,
+		sleep(50); // the direct delay is needed to force it back into the proper gamemode. it cannot for the life of it do so at the exact same moment and requires the delay.
+		for([str('gamemode %s %s', e:'gm', e:'name')],
+			logger('warn', _);
+			run(_);
+		);
```

Comment:
For running this single command a loop is not neccesary, and in general, I don't really see a need to also log the command. In my opinion it would suffice to use `modify(..., 'gamemode', ...)`